tools/topology/m4/tdfb_controls.m4

```diff
@@ -0,0 +1,33 @@
+# TDFB Bytes control with max value of 255
+C_CONTROLBYTES(DEF_TDFB_BYTES, PIPELINE_ID,
+	CONTROLBYTES_OPS(bytes,
+		258 binds the mixer control to bytes get/put handlers,
+		258, 258),
+	CONTROLBYTES_EXTOPS(258 binds the mixer control to bytes get/put handlers,
+		258, 258),
+	, , ,
+	CONTROLBYTES_MAX(, 4096),
+	,
+	DEF_TDFB_PRIV)
+
+# TDFB enum list
+CONTROLENUM_LIST(DEF_TDFB_ONOFF, LIST(`	', `"beam on"', `"beam off"'))
+
+# TDFB enum control
+C_CONTROLENUM(DEF_TDFB_ONOFF_ENUM, PIPELINE_ID,
+	DEF_TDFB_ONOFF,
+	LIST(`	', ENUM_CHANNEL(FC, 3, 0)),
+	CONTROLENUM_OPS(enum,
+		257 binds the mixer control to enum get/put handlers,
+		257, 257))
+
+# TDFB enum list
+CONTROLENUM_LIST(DEF_TDFB_BEAMN, LIST(`	', `"0"', `"30"', `"60"', `"90"', `"120"', `"150"', `"180"', `"210"', `"240"', `"270"', `"300"', `"330"'))
```

Comment:
I shortened the control names to plain numbers without "deg". I like myself the -180 ... +180 notation more but this way the enum can be converted to angle with multiplication and power-on default enum 0 remains 0 degrees. The multiplier and offset for enum to angle conversion is passed in binary configuration blob. So the -180 .. +180 would be possible to achieve with offset of 90 degrees.  For line array the control range is -90 ... +90. With the all-positive angles it would be 0..90 and 270..330. Changing this is on topology and tools side if seen better.

Reply:
Same, the firmware supports with configuration blob (multiplier and offset for angle) various angle to enum codings. The tools side needs work to make it easier. Now there's manual script editing needed to change the angles coding.